### ISSUE TITLE
Any way to avoid "x: 0.00001" to "x=1.0E-5" 

### ISSUE THREAD

**Issue by cndaimin** [lightbend/config#581]
My code:
`public static void main(String[] args) throws Exception {
        Config conf = ConfigFactory.parseString("x: 0.00001");
        String rendered = conf.resolve().root().render();
        System.out.println(rendered);
    }`

output is: 
{
    # String: 1
    "x" : 1.0E-5
}
I prefer 0.00001 rather than 1.0E-5, so my question is how to prevent the transformation

**Comment by havocp** [comment]
The original text is known and used here: https://github.com/lightbend/config/blob/40d0c40846940e3780cb5741610182341f6b7c5d/config/src/main/java/com/typesafe/config/impl/ConfigNumber.java#L30-L33

But when rendering we convert to a double and use Double.toString here:
 https://github.com/lightbend/config/blob/40d0c40846940e3780cb5741610182341f6b7c5d/config/src/main/java/com/typesafe/config/impl/AbstractConfigValue.java#L359-L362

Overriding that render method in `ConfigNumber` to use `originalText` might work, as long as all originalText is valid JSON (I think it would be... but first question is whether tests still pass with this) 

Note that there's no overall goal to preserve formatting when rendering, so in a case like this where it's easy maybe we want to do it, but any case where it's hard we would not. There's a separate not-very-complete API included called ConfigDocument to load docs while preserving the formatting.

**Comment by Emurus** [comment]
Is it possible to make transformToString public to get the original text?

**Comment by Emurus** [comment]
Rather than return originalText itself, why not return transformToString()?

**Comment by havocp** [comment]
right, I mean that may be the detail of it. I haven’t dug in on it to see what might break.
